merkle/map_hasher.go

```diff
@@ -0,0 +1,41 @@
+package merkle
+
+import (
+	"github.com/google/trillian"
+)
+
+// MapHasher is a specialised TreeHasher which also knows about the set of
+// "null" hashes for the unused sections of a SparseMerkleTree.
+type MapHasher struct {
```

Comment:
Does this line up with the E2E map hashing? IIRC he's using tree level explicitly somewhere.